src/PointwiseFunctions/AnalyticData/Burgers/Periodic.hpp

```diff
@@ -0,0 +1,115 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/DataBox/Prefixes.hpp"  // IWYU pragma: keep
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Evolution/Systems/Burgers/Tags.hpp"  // IWYU pragma: keep
+#include "Options/Options.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+/// \cond
+class DataVector;
+// IWYU pragma: no_forward_declare Tensor
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace Burgers {
+namespace AnalyticData {
+/*!
+ * \brief Analytic data (with an "exact" solution known) that is periodic over
+ * the interval \f$[0,2\pi]\f$.
+ *
+ * The initial data is given by:
+ *
+ * \f{align}{
+ *   u(x, 0) = \sin(x)
+ * \f}
+ *
+ * At future times the analytic solution can be found by solving the
+ * transcendental equation \cite HARTEN19973
+ *
+ * \f{align}{
+ *   \label{eq:transcendental burgers periodic}
+ *   \mathcal{F}=\sin\left(x-\mathcal{F}t\right)
+ * \f}
+ *
+ * on the interval \f$x\in(0,\pi)\f$. The solution from \f$x\in(\pi,2\pi)\f$ is
+ * given by \f$\mathcal{F}(x, t)=-\mathcal{F}(2\pi-x,t)\f$. The transcendental
+ * equation \f$(\ref{eq:transcendental burgers periodic})\f$ can be solved with
+ * a Newton-Raphson iterative scheme. Since this can be quite sensitive to the
+ * initial guess we implement this solution as analytic data. The python code
+ * below can be used to compute the analytic solution if desired.
+ *
+ * At time \f$1\f$ the solution develops a discontinuity at \f$x=\pi\f$ followed
+ * by the amplitude of the solution decaying over time.
+ *
+ * \note We have rescaled \f$x\f$ and \f$t\f$ by \f$\pi\f$ compared to
+ * \cite HARTEN19973.
+ *
+ * \code{py}
+   def burgers_periodic(x_grid, time):
```

Comment:
What would you think of editing this snippet slightly to make it easier to use, by adding the needed imports and documenting the format of `x_grid`? For example, by adding these lines
```
import numpy as np
from scipy.optimize import newton
# x_grid is a np.array of positions at which to evaluate the solution
```